capstone/src/main/webapp/script.js

```diff
@@ -287,44 +290,55 @@ async function getInputFilters() {
 
   // do manual search if filter is selected, else do Nearby Search w Places API
   if (_showSmallBusiness || _showBlackOwnedBusiness) {
-    if (!isStringEmpty(keyword)) {
-      const keywordEntities = getEntities(keyword);
-      // TODO(#47): manual search
-    }
+    await(manualSearch(keyword));
   } else {
     await getPlacesSearchResults(keyword);
   }
+  _showSmallBusiness = false;
+  _showBlackOwnedBusiness = false;
   document.getElementById('search-button').disabled = false;
 }
 
 /** @param {String} str input string to check
-    @return {boolean} if str is empty, contains only white space, or null */
+    @return {boolean} if str is empty, contains only white space, or null 
+ */
 function isStringEmpty(str) {
   return (str.length === 0 || !str.trim() || !str);
 }
 
-async function getReviewsEntities(reviews) {
-  // TODO(#33): integrate with actual reviews of businesses
-  const reviewsEntities = await getEntities(reviews);
-  return reviewsEntities;
+/** @returns{Object} gets top 20 results, sorted by avg review,
+    matching @param {String} keyword 
+ */
+async function manualSearch(keyword) {
```

Comment:
I could be wrong, but I don't think this function needs the async keyword. You're returning a Promise and aren't using `await` within the function, so `async` should be unnecessary.

Reply:
the reason why I made this function async is because I want to wait for this function to finish returning results before re-enabling the search button. I'm not super familiar with async, but I thought this would be needed to make sure those events occur synchronously?

Reply:
Ahh yeah so async\await is confusing.

`await` is used to block on a promise. Like
```
... // do things
await myPromise;
... // do more things
```
This can be normal promises, or functions with the `async` keyword.

You only need the `async` keyword on a function if there's an `await` inside of it. It just tells the javascript interpreter that "hey, treat this like a promise because there's something asynchronous inside of it."

So for your use case, since there's no `await` inside the function, you don't need it. You're returning a promise from `manualSearch()`, so the `await` will work, since it `await`s the promise that `manualSearch()` returns

Reply:
ahh that makes sense thanks!